### PR TITLE
[nrf toup] Add UART flow control support

### PR DESCRIPTION
**Pull request by joerchan**
Add UART flow control for nordic platform UART driver.
Add generic platform support for respecting platform specific UART configuartions.

### CODE REVIEW

File: platform/ext/target/nordic_nrf/common/core/cmsis_drivers/Driver_USART.c [hunk 1]
```diff
@@ -323,7 +323,7 @@ static ARM_USART_MODEM_STATUS ARM_USART_GetModemStatus(void)
         .baudrate = NRF_UARTE_BAUDRATE_115200,                            \
         .interrupt_priority = NRFX_UARTE_DEFAULT_CONFIG_IRQ_PRIORITY,     \
         .hal_cfg  = {                                                     \
-            .hwfc   = NRF_UARTE_HWFC_DISABLED,                            \
+            .hwfc = RTE_USART##idx##_HWFC,                                \
```
Comment (Vge0rge) [inline platform/ext/target/nordic_nrf/common/core/cmsis_drivers/Driver_USART.c:326]: nit: Make the = value aligned with the following lines.

File: platform/ext/target/nordic_nrf/nrf5340dk_nrf5340_cpuapp/RTE_Device.h [hunk 2]
```diff
@@ -80,6 +86,8 @@
 //   </h> Pin Configuration
 // </e> USART (Universal synchronous - asynchronous receiver transmitter) [Driver_USART3]
 
+#define   RTE_USART2_HWFC               NRF_UARTE_HWFC_DISABLED
```
Comment (Vge0rge) [inline platform/ext/target/nordic_nrf/nrf5340dk_nrf5340_cpuapp/RTE_Device.h:89]: Is that not meant to be USART3_HWFC?

File: platform/ext/target/nordic_nrf/nrf9160dk_nrf9160/RTE_Device.h [hunk 3]
```diff
@@ -80,6 +86,7 @@
 //   </h> Pin Configuration
 // </e> USART (Universal synchronous - asynchronous receiver transmitter) [Driver_USART3]
 
+#define   RTE_USART2_HWFC               NRF_UARTE_HWFC_DISABLED
```
Comment (Vge0rge) [inline platform/ext/target/nordic_nrf/nrf9160dk_nrf9160/RTE_Device.h:89]: Same as before.

**Review by Vge0rge — APPROVED** [review]

**Review by SebastianBoe — CHANGES_REQUESTED** [review]
toup is not allowed IIRC

**Review by SebastianBoe — APPROVED** [review]

**Comment by joerchan** [comment]
> toup is not allowed IIRC

This was temporary, the patches have been submitted upstream, changed to fromlist.

**Pull request merged**